flake.nix

```diff
@@ -1,12 +1,24 @@
 {
   description = "binplz";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-22.05";
-  inputs.flake-utils.url = "github:numtide/flake-utils";
-  inputs.nixos-channel.url = "https://nixos.org/channels/nixos-22.05/nixexprs.tar.xz";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    # This URL isn't stable, so the first time you check out this repo,
+    # all developers need to run `nix flake lock --update-input nixos-channel`.
+    # See https://github.com/binplz/binplz.dev/issues/5
+    nixos-channel.url = "https://nixos.org/channels/nixos-22.05/nixexprs.tar.xz";
+    nixos-generators.url = "github:nix-community/nixos-generators";
+    nixos-generators.inputs.nixpkgs.follows = "nixpkgs";
+  };
 
   outputs = inputs:
     let
+
+      system = "x86_64-linux";
```

Comment:
Why not `eachSystem`?

https://github.com/numtide/flake-utils#eachsystem---system---system---attrs

With it, we can keep the `perSystem` style of description.

Reply:
It was annoying during development because I was initially working with `nixos-rebuild build-vm`, which made it more annoying. I'll see if I can change it back to `perSystem`.

Reply:
Oh, that is annoying. I just thought it would be nicer if we could keep `perSystem` style. But I don't have a strong opinion over it. If it is hard to get back, please leave it.